src/azure-cli-telemetry/azure/cli/telemetry/util.py

```diff
@@ -4,41 +4,8 @@
 # --------------------------------------------------------------------------------------------
 
 import os
-import stat
 import logging
 import logging.handlers
-from datetime import datetime
-
-from azure.cli.telemetry.const import TELEMETRY_NOTE_NAME, MANDATORY_WAIT_PERIOD
-
-
-def should_upload(config_dir):
-    """Returns True if it is the right moment to add telemetry.
-    Before the client request a telemetry add process, it can invoke this method to test if it is the right moment.
-    The conditions are:
-        1. The telemetry.txt file doesn't exist; OR
-        2. The telemetry.txt file is a regular file AND there have been enough time passed since last add.
-    """
-    logger = logging.getLogger('telemetry.check')
-
-    telemetry_note_path = os.path.join(config_dir, TELEMETRY_NOTE_NAME)
-    if not os.path.exists(telemetry_note_path):
-        logger.info('Positive: The %s does not exist.', telemetry_note_path)
-        return True
-
-    file_stat = os.stat(telemetry_note_path)
-    if not stat.S_ISREG(file_stat.st_mode):
-        logger.warning('Negative: The %s is not a regular file.', telemetry_note_path)
-        return False
-
-    modify_time = datetime.fromtimestamp(file_stat.st_mtime)
-    if datetime.now() - modify_time < MANDATORY_WAIT_PERIOD:
```

Comment:
On a second thought, would it be safer to keep the existing code and let `MANDATORY_WAIT_PERIOD = 0` in order to make the wait period configurable?